crates/trie/sparse-parallel/src/trie.rs

```diff
@@ -636,11 +645,12 @@ struct ChangedSubtrie {
     prefix_set: PrefixSet,
 }
 
-/// Convert first two nibbles of the path into a lower subtrie index in the range [0, 255].
+/// Convert first [`UPPER_TRIE_MAX_DEPTH`] nibbles of the path into a lower subtrie index in the
+/// range [0, [`NUM_LOWER_SUBTRIES`]).
 ///
 /// # Panics
 ///
-/// If the path is shorter than two nibbles.
+/// If the path is shorter than [`UPPER_TRIE_MAX_DEPTH`] nibbles.
 fn path_subtrie_index_unchecked(path: &Nibbles) -> usize {
```

Comment:
`path_subtrie_index_unchecked` doesn't work with variable size max depth, but I think it doesn't worth it to make it dynamic. Let's just add an assertion.
```suggestion
fn path_subtrie_index_unchecked(path: &Nibbles) -> usize {
    debug_assert_eq!(UPPER_TRIE_MAX_DEPTH, 2);
```

Reply:
Good catch :+1: